hedera-node/hedera-app/src/main/java/com/hedera/node/app/tss/TssBaseServiceImpl.java

```diff
@@ -146,27 +148,34 @@ public void setCandidateRoster(@NonNull final Roster roster, @NonNull final Hand
         final var candidateRosterParticipantDirectory = computeTssParticipantDirectory(roster, maxSharesPerNode, (int)
                 context.networkInfo().selfNodeInfo().nodeId());
 
-        final AtomicInteger shareIndex = new AtomicInteger(0);
-        for (final var tssPrivateShare : tssPrivateShares) {
-            final var tssMsg = CompletableFuture.supplyAsync(
-                            () -> tssLibrary.generateTssMessage(candidateRosterParticipantDirectory, tssPrivateShare),
-                            tssLibraryExecutor)
-                    .exceptionally(e -> {
-                        log.error("Error generating tssMessage", e);
-                        return null;
-                    });
-            tssMsg.thenAccept(msg -> {
-                if (msg == null) {
-                    return;
-                }
-                final var tssMessage = TssMessageTransactionBody.newBuilder()
-                        .sourceRosterHash(activeRosterHash)
-                        .targetRosterHash(candidateRosterHash)
-                        .shareIndex(shareIndex.getAndAdd(1))
-                        .tssMessage(Bytes.wrap(msg.bytes()))
-                        .build();
-                tssSubmissions.submitTssMessage(tssMessage, context);
-            });
+        // https://github.com/hashgraph/hedera-services/issues/16166
+        final boolean hasKeyMaterial = false;
+        if (!hasKeyMaterial) {
+            final AtomicInteger shareIndex = new AtomicInteger(0);
+            for (final var tssPrivateShare : tssPrivateShares) {
+                final var tssMsg = CompletableFuture.supplyAsync(
+                                () -> tssLibrary.generateTssMessage(
+                                        candidateRosterParticipantDirectory, tssPrivateShare),
+                                tssLibraryExecutor)
+                        .exceptionally(e -> {
+                            log.error("Error generating tssMessage", e);
+                            return null;
+                        });
+                tssMsg.thenAccept(msg -> {
+                    if (msg == null) {
+                        return;
+                    }
+                    final var tssMessage = TssMessageTransactionBody.newBuilder()
+                            .sourceRosterHash(activeRosterHash)
+                            .targetRosterHash(candidateRosterHash)
+                            .shareIndex(shareIndex.getAndAdd(1))
+                            .tssMessage(Bytes.wrap(msg.bytes()))
+                            .build();
+                    tssSubmissions.submitTssMessage(tssMessage, context);
+                });
+            }
+        } else {
+
```

Comment:
Why is this else empty ?

Reply:
I forgot to add a link to the issue that will implement this, thanks!

Reply:
I ended up getting rid of the if/else; it's more confusing than helpful. We can implement it in another issue, and just focus on happy path right now